### DISCUSSION TITLE
mosaic map endpoint help

### DISCUSSION THREAD

**Discussion by hrodmn** [developmentseed/eoAPI#74] [Q&A]
I have a mosaic `searchid` registered (one collection, no bounding box or datetime filter) and can load the XYZ tiles into QGIS or a Leaflet map, but the `map` endpoint doesn't seem to be behaving for me. Am I using it wrong? When I open it I get a mostly blank map canvas:

![image](https://github.com/developmentseed/eoAPI/assets/17726304/4fc8cbc2-7bc4-4ae9-b9ad-fdc599c7bd5c)

Here is what my map endpoint url looks like for a composite of three assets:
`{raster_endpoint}/mosaic/{searchid}/map?assets=120&assets=900&assets=800&rescale=0,0.6&rescale=0,0.6&rescale=0,0.6`

The XYZ tile url that works:
`{raster_endpoint}/mosaic/{searchid|/tiles/WebMercatorQuad/{z}/{x}/{y}@1x?assets=120&assets=900&assets=800&rescale=0,0.6&rescale=0,0.6&rescale=0,0.6`

**Comment by vincentsarago** [comment]
@hrodmn 
- is there any logs in the frontend? 
- is the endpoint local or deployed?
- maybe there is a IAM role issue (maybe the data is not available to the tiler)
- if the data is not in one of your bucket (or not in the same region), the tiler won't have access to it (https://github.com/developmentseed/eoAPI/pull/60)

**Reply by hrodmn** [reply]
> * is there any logs in the frontend?

I don't see any error messages when I open the map in my browser or in the raster Lambda function logs.

> * is the endpoint local or deployed?

The endpoint is deployed on AWS us-east-1.

> * maybe there is a IAM role issue (maybe the data is not available to the tiler)
> * if the data is not in one of your bucket (or not in the same region), the tiler won't have access to it ([update Subnets #60](https://github.com/developmentseed/eoAPI/pull/60))

I am able to browse the tiles with the tileJSON XYZ tile URL so I think the tiler can access the data. The raster STAC item viewer is working great, too!

Maybe it would be easier to debug it if I run it locally?

**Reply by vincentsarago** [reply]
> Maybe it would be easier to debug it if I run it locally?

maybe, at least it could tell if this is am IAM issue

**Reply by vincentsarago** [reply]
Oh sorry I answered too fast 😀 

So if you are able to see the tiles in the item endpoint it means it's mostly a frontend issue

**Reply by vincentsarago** [reply]
The problem with mosaic is usually the zoom definition, it might be wise to set the min/max zoom in the mosaic metadata or in the tilejson call

**Reply by hrodmn** [reply]
I fired up a local docker network pointing at the deployed database and got some actual error logs:

```sh
eoapi.raster  |   warnings.warn(
eoapi.raster  | 172.24.0.1:37482 - "GET /mosaic/699a47ff8d030659daa73f78b9ec3410/tiles/WebMercatorQuad/2/-4/3%401x?assets=120&rescale=0%2C1 HTTP/1.1" 404
eoapi.raster  | /opt/bitnami/python/lib/python3.11/site-packages/morecantile/models.py:455: PointOutsideTMSBounds: Point (50093770.85697311, -20037508.342789244) is outside TMS bounds [-20037508.342789244, -20037508.342789244, 20037508.342789244, 20037508.342789244].
```

**Reply by vincentsarago** [reply]
> `/WebMercatorQuad/2/-4/3`

The tile index is weird 🤔 , maybe the bbox in the mosaic is wrong!

**Reply by hrodmn** [reply]
I think it wasn't working because I did not set a value for `bbox` when I registered the searchid. A new searchid with `bbox` set correctly fixes the problem. Maybe we could add a default `bbox=[-180, -90, 180, 90]` at some point upstream?

**Reply by vincentsarago** [reply]
🤔 there should be a default bbox https://github.com/stac-utils/titiler-pgstac/blob/main/titiler/pgstac/factory.py#L408-L411
